### PR TITLE
1743 support emit instruc

### PR DESCRIPTION
**Pull request by zainen**
Adds the `EMIT` instruction to the retStack function

adds two passing test and one invalid contract test 

current unknowns if done correctly are:

how is the type persisted to be added with the Event

when the `CONS` instruction is added to the stack it puts the `operation` into a list but loses the annotation.

is there a place missed to add the instruction `ty` as well as the annotated `tag` ? 

Before submitting this PR, please make sure:

- [ ] Your code builds cleanly without any errors or warnings
- [ ] You have added unit tests
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: packages/taquito-michel-codec/test/contracts_014.spec.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  ]
+};
+
+describe('PtKathmandu', () => {
+  for (const [group, list] of Object.entries(contracts)) {
+    describe(group, () => {
+      for (const contract of list) {
+        it(contract, () => {
+          const options: ContractOptions = {
+            protocol: Protocol.PtJakarta,
```
Comment (roxaneletourneau) [inline packages/taquito-michel-codec/test/contracts_014.spec.ts:35]: Should be protocol Kathmandu

Comment (zainen) [reply packages/taquito-michel-codec/test/contracts_014.spec.ts:35]: 👍

File: packages/taquito-michel-codec/src/michelson-typecheck.ts [hunk 2]
```diff
@@ -2004,9 +2004,20 @@ function functionTypeInternal(
         ensurePushableType(s[0]);
         return [annotateVar({ prim: 'option', args: [instruction.args[1]] }), ...stack.slice(2)];
       }
+
       case 'MIN_BLOCK_TIME':
         return [annotateVar({ prim: 'nat' }), ...stack];
 
+      case 'EMIT': {
+        const ia = instructionAnn({ f: 1, t: 1}, { specialVar: true });
```
Comment (roxaneletourneau) [inline packages/taquito-michel-codec/src/michelson-typecheck.ts]: consider removing `{ specialVar: true }` as it doesn't apply to EMIT

**Review by roxaneletourneau — COMMENTED** [review]
Looks like `EMIT` should be in `primitives` in the binary.ts file

**Review by roxaneletourneau — APPROVED** [review]

**Pull request merged**